packages/esm-patient-medications-app/src/medications-summary/order-basket-side-rail.component.tsx

```diff
@@ -0,0 +1,44 @@
+import React, { useState, useMemo } from 'react';
+import { launchPatientWorkspace, ScreenModeTypes, WindowSize } from '@openmrs/esm-patient-common-lib';
+import { Button } from 'carbon-components-react';
+import styles from './order-basklet-side-rail.scss';
+import ShoppingCart20 from '@carbon/icons-react/es/shopping--cart/20';
+import WarningFilled16 from '@carbon/icons-react/es/warning--filled/16';
+import Close20 from '@carbon/icons-react/es/close/20';
+import { useAssignedExtensionIds } from '@openmrs/esm-framework';
+
+interface OrderBasketSideRailProps {
+  windowSize: WindowSize;
+  checkViewMode: (active: boolean) => void;
+}
+
+const OrderBasketSideRail: React.FC<OrderBasketSideRailProps> = ({ windowSize, checkViewMode }) => {
+  const extensions = useAssignedExtensionIds('patient-chart-workspace-slot');
+  const [mouseHover, setMouseHover] = useState<boolean>(false);
+
+  const active = useMemo(() => extensions.some((ext) => ext === 'order-basket-workspace'), [extensions]);
+
+  const handleClick = () => (active ? checkViewMode(active) : launchPatientWorkspace('order-basket-workspace'));
```

Comment:
Isn't this the behavior we always (or usually) want from workspaces—"if it's already open, go to that tab, otherwise open a new tab"? Can we just put this logic in `launchPatientWorkspace`?

Relatedly, I would like for us to think about the workspace as its own abstraction, which has nothing to do with extensions, but happens to use the extension system under the hood. So someone writing code which interacts with workspaces should just be able to use functions from `common-lib`, like `openWorkspaceTab` and `launchPatientWorkspace`, and not have to know anything about the extension system.